src/commands/deployWorkspaceProject/deploymentConfiguration/workspace/FilePathsVerifyStep.ts

```diff
@@ -3,50 +3,71 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { nonNullProp } from "@microsoft/vscode-azext-utils";
+import { AzExtFsExtra, GenericTreeItem, activityFailContext, activityFailIcon, activitySuccessContext, activitySuccessIcon, nonNullProp, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
 import * as path from "path";
 import { type Progress } from "vscode";
 import { ExecuteActivityOutputStepBase, type ExecuteActivityOutput } from "../../../../utils/activity/ExecuteActivityOutputStepBase";
+import { createActivityChildContext } from "../../../../utils/activity/activityUtils";
 import { localize } from "../../../../utils/localize";
 import { type WorkspaceDeploymentConfigurationContext } from "./WorkspaceDeploymentConfigurationContext";
 
-export class FilePathsVerifyStep extends ExecuteActivityOutputStepBase<WorkspaceDeploymentConfigurationContext> {
-    public priority: number = 190;  /** Todo: Figure out a good priority level */
+export abstract class FilePathsVerifyStep extends ExecuteActivityOutputStepBase<WorkspaceDeploymentConfigurationContext> {
+    abstract key: string;
+    abstract fileType: string;
+
+    private configPath: string | undefined;
+
+    public constructor() {
+        super();
+    }
 
-    // Todo: Add logic to verify that the file paths actually exist, if they don't, leave the corresponding context value as undefined
     protected async executeCore(context: WorkspaceDeploymentConfigurationContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
         this.options.shouldSwallowError = true;
-        progress.report({ message: localize('verifyingFilePaths', 'Verifying file paths...') });
+        progress.report({ message: localize('verifyingFilePaths', `Verifying file paths...`) });
 
         const rootPath: string = nonNullProp(context, 'rootFolder').uri.fsPath;
 
-        const configDockerfilePath = context.deploymentConfigurationSettings?.dockerfilePath;
-        if (!context.dockerfilePath && configDockerfilePath) {
-            context.dockerfilePath = path.join(rootPath, configDockerfilePath);
-        }
+        this.configPath = nonNullValueAndProp(context.deploymentConfigurationSettings, this.key as keyof typeof context.deploymentConfigurationSettings)
```

Comment:
```suggestion
        this.configPath = nonNullValueAndProp(context.deploymentConfigurationSettings, this.key)
```

I think we can add in built-in type safety so we don't need to type cast this.  Here's what you can do:

`abstract key: keyof DeploymentConfigurationSettings;`

Then you can just reference the implementing class' property like this:

`key = 'envPath' as const;` and it'll check for type safety automatically!

I think I could incorporate something similar into the Azure resource PR as well so that'll be a nice addition